src/pages/admin/student/edit/page.tsx

```diff
@@ -1,17 +1,33 @@
 import * as React from 'react';
-import {Link} from 'react-router';
-import {adminRouteEnums} from '../../../../common/routeEnums/admin';
+import { Link } from 'react-router';
+import { adminRouteEnums } from '../../../../common/routeEnums/admin';
+import { StudentSummary } from '../../../../model/studentSummary';
 
-export class EditStudentPage extends React.Component<{}, {}> {
-   public render() {
+interface Props  {
+  student: StudentSummary;
+  getstudent : (id: number) => void;
+  params? : any;
+}
+
+export class EditStudentPage extends React.Component<Props, {}> {
+   componentDidMount() {
+       const studentId: number = Number(this.props.params.id);
```

Comment:
I think EditStudentPage should receive studentId as props instead of params to be agnostic of router params. You can take a look how is implemented in:

#### ./src/pages/trainer/training/edit/pageContainer.tsx
and
#### ./src/pages/trainer/training/edit/page.tsx